.gitmodules

```diff
@@ -268,6 +268,13 @@
 [submodule "contrib/hashidsxx"]
 	path = contrib/hashidsxx
 	url = https://github.com/schoentoon/hashidsxx.git
+[submodule "contrib/udns"]
+	path = contrib/udns
+	url = https://github.com/arthurpassos/udns.git
```

Comment:
(Upstream development of UDNS seemingly died somewhen in 2014 and there are also only tarballs for download on the corresponding website. I found https://github.com/ortclib/udns which is an older import of the source into a GitHub repository. I am really not against referencing your fork here, I just wonder if we could perhaps use the existing fork?)

Reply:
Of course we cannot use any of these forks (https://github.com/arthurpassos/udns.git, https://github.com/ortclib/udns or any other non-official repo with 3.5 stars created by noname user). We have to make our own fork in ClickHouse organization if really need this library.

Reply:
Thanks for the clarification. I suppose there is a risk that if the owner of a repo with too few stars/forks decides to take down the repo while ClickHouse references it, our build will break.

So, here you go:
- https://github.com/ClickHouse/udns
- https://github.com/ClickHouse/cpp-dns

Reply:
Btw, it makes sense to review the content of `udns` repo, because it's not clear where the code was taken from. And there are no guarantees that code of the original library was not modified.

(sorry for being that paranoid, but I don't like how the repo with one commit named "everything" looks)

Reply:
Sure, feel free to do it! Let me know if I can help